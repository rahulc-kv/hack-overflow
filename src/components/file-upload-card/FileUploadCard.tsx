import React, { FC } from 'react';
import { Tooltip } from '@mui/material';

import {
  GreenTickFillIcon,
  CloseIcon,
  RefreshIconLite,
  CrossIconRoundRed
} from '@assets/icons';
import { FILE_UPLOAD_STATUS } from '@constants/common';
import { translate } from '@utils/locale';
import { FileUploadCardProps } from './types';
import COLORS from '@constants/colors';

const FileUploadCard: FC<FileUploadCardProps> = props => {
  const { id, fileName, status, handleClose } = props;

  const renderContent = () => {
    switch (status) {
      case FILE_UPLOAD_STATUS.success:
        return (
          <div className="flex flex-row justify-between items-center text-sm font-normal text-slateGrey">
            <div className="flex flex-row w-[90%]">
              <div className="mr-[5px] w-[22px] h-[22px]">
                <GreenTickFillIcon width={22} height={22} />
              </div>
              <Tooltip title={fileName} arrow={true} enterTouchDelay={0}>
                <div className="truncate">{fileName}</div>
              </Tooltip>
            </div>
            <CloseIcon
              stroke={COLORS.DAVY_GREY}
              className="cursor-pointer"
              onClick={() => handleClose(id, status)}
              data-testid="closeSuccess"
            />
          </div>
        );
      case FILE_UPLOAD_STATUS.uploading:
        return (
          <div className="flex flex-row justify-between items-center text-sm font-normal text-slateGrey">
            <div className="flex flex-row w-[90%]">
              <div
                className="flex justify-center items-center p-1 mr-[5px] w-[21px] h-[21px]
                bg-slateGrey rounded-[50%]">
                <RefreshIconLite
                  width={11}
                  height={11}
                  className="animate-rotate"
                />
              </div>
              <Tooltip title={fileName} arrow={true} enterTouchDelay={0}>
                <div className="truncate">{fileName}</div>
              </Tooltip>
            </div>
            <CloseIcon
              stroke={COLORS.DAVY_GREY}
              className="cursor-pointer"
              onClick={() => handleClose(id, status)}
              data-testid="closeUploading"
            />
          </div>
        );
      case FILE_UPLOAD_STATUS.failed:
        return (
          <div className="flex flex-row justify-between items-center w-full text-sm font-normal text-slateGrey">
            <div className="flex flex-row w-[70%]">
              <div className="mr-[5px] w-[22px] h-[22px]">
                <CrossIconRoundRed width={22} height={22} />
              </div>
              <Tooltip title={fileName} arrow={true} enterTouchDelay={0}>
                <div className="truncate">{fileName}</div>
              </Tooltip>
            </div>
            <div className="flex flex-row items-center">
              <div
                className="py-[1px] px-[9px] mr-1 min-w-[50px] text-[11px] font-normal 
              text-slateGrey bg-harp rounded-[100px]">
                {translate('sourcePage.failed')}
              </div>
              <CloseIcon
                stroke={COLORS.DAVY_GREY}
                className="cursor-pointer"
                onClick={() => handleClose(id, status)}
                data-testid="closeFailed"
              />
            </div>
          </div>
        );
    }
  };
  return (
    <div className="py-2 px-[10px] rounded-md border-[1px] border-greyCloud">
      {renderContent()}
    </div>
  );
};

export default FileUploadCard;
