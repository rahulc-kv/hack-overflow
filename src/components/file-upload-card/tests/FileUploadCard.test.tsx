import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import FileUploadCard from '../FileUploadCard';

const handleUploadAbort = jest.fn();

test('FileUploadCard rendered', () => {
  const component = render(
    <FileUploadCard
      id={'asfdsds'}
      fileName={'report'}
      status={'success'}
      handleClose={handleUploadAbort}
    />
  );
  expect(component).toMatchSnapshot();
});

test('FileUploadCard Success Close', () => {
  const { getByTestId } = render(
    <FileUploadCard
      id={'ashjkfsd'}
      fileName={'transactionReport'}
      status={'success'}
      handleClose={handleUploadAbort}
    />
  );
  fireEvent.click(getByTestId('closeSuccess'));
  expect(handleUploadAbort).toHaveBeenCalled();
});

test('FileUploadCard Uploading Close', () => {
  const { getByTestId } = render(
    <FileUploadCard
      id={'testid'}
      fileName={'transactionReport'}
      status={'uploading'}
      handleClose={handleUploadAbort}
    />
  );
  fireEvent.click(getByTestId('closeUploading'));
  expect(handleUploadAbort).toHaveBeenCalled();
});

test('FileUploadCard Failed Close', () => {
  const { getByTestId } = render(
    <FileUploadCard
      id={'testid'}
      fileName={'transactionReport'}
      status={'failed'}
      handleClose={handleUploadAbort}
    />
  );
  fireEvent.click(getByTestId('closeFailed'));
  expect(handleUploadAbort).toHaveBeenCalled();
});
