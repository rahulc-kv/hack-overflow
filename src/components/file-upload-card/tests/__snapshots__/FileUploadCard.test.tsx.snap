// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploadCard rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-2 px-[10px] rounded-md border-[1px] border-greyCloud"
      >
        <div
          class="flex flex-row justify-between items-center text-sm font-normal text-slateGrey"
        >
          <div
            class="flex flex-row w-[90%]"
          >
            <div
              class="mr-[5px] w-[22px] h-[22px]"
            >
              <svg
                height="22"
                width="22"
              >
                greenTickFillIcon.svg
              </svg>
            </div>
            <div
              aria-label="report"
              class="truncate"
              data-mui-internal-clone-element="true"
            >
              report
            </div>
          </div>
          <svg
            class="cursor-pointer"
            data-testid="closeSuccess"
            stroke="#4B5563"
          >
            closeIcon.svg
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-2 px-[10px] rounded-md border-[1px] border-greyCloud"
    >
      <div
        class="flex flex-row justify-between items-center text-sm font-normal text-slateGrey"
      >
        <div
          class="flex flex-row w-[90%]"
        >
          <div
            class="mr-[5px] w-[22px] h-[22px]"
          >
            <svg
              height="22"
              width="22"
            >
              greenTickFillIcon.svg
            </svg>
          </div>
          <div
            aria-label="report"
            class="truncate"
            data-mui-internal-clone-element="true"
          >
            report
          </div>
        </div>
        <svg
          class="cursor-pointer"
          data-testid="closeSuccess"
          stroke="#4B5563"
        >
          closeIcon.svg
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
